src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,120 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ARRAYCOPY_POSITION = 0;
+    private static final int ARRAYCOPY_MOVE_POSITION = 1;
+    private static final double GROW_SIZE = 1.5;
+    private static final String INDEX_EXCEPTION_MESSAGE = "ArrayList index out of bounds";
+    private static final String NO_SUCH_ELEMENT_EXCEPTION_MESSAGE = "No such element";
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (isFull()) {
+            grow();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (isFull()) {
+            grow();
+        }
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(INDEX_EXCEPTION_MESSAGE);
+        }
+        Object indexPositionElement = elements[index];
+        elements[index] = value;
+        size++;
+        Object tempElement;
+        for (int i = index + 1; i < size; i++) {
+            tempElement = elements[i];
+            elements[i] = indexPositionElement;
+            indexPositionElement = tempElement;
+        }
```

Comment:
Use System.arraycopy. It's more optimized
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        checkIndex(index);
        if (isFull()) {
            grow();
        }
        System.arraycopy(elements, index, elements, index + 1, size - index);
        elements[index] = value;
        size++;
```